src/Compilers/Core/Portable/Compilation/SyntaxTreeOptionsProvider.cs

```diff
@@ -14,9 +14,14 @@ public abstract class SyntaxTreeOptionsProvider
         public abstract bool? IsGenerated(SyntaxTree tree);
 
         /// <summary>
-        /// Get diagnostic severity setting or a given diagnostic identifier in a given tree.
+        /// Get diagnostic severity setting for a given diagnostic identifier in a given tree.
         /// </summary>
         public abstract bool TryGetDiagnosticValue(SyntaxTree tree, string diagnosticId, out ReportDiagnostic severity);
+
+        /// <summary>
+        /// Get diagnostic severity set globally for a given diagnostic identifier
+        /// </summary>
+        public abstract bool TryGetGlobalDiagnosticValue(string diagnosticId, out ReportDiagnostic severity);
```

Comment:
>TryGetGlobalDiagnosticValue [](start = 29, length = 27)

Sorry if this was already discussed with feature crew offline, but what consumer of our public API needs this new API? 
I would have thought the existing `TryGetDiagnosticValue` would be sufficient.

Reply:
Because global options can exist 'ambiently' without applying to any specific tree. One option would be to call `TryGetDiagnosticValue` without a `null` tree but that seems icky.



---
In reply to: [471710188](https://github.com/dotnet/roslyn/pull/45871#discussion_r471710188) [](ancestors = 471710188)